### PR TITLE
Implement heartbeat queue

### PR DESCRIPTION
**Pull request by Otto-AA**
Regarding #3 

- [x] Implement heartbeat queue
- [ ] Test heartbeat queue

I've added a test case which should test the queue, but it seems that the current version without queue also fulfills this test.

### CODE REVIEW

**Comment by johan-bjareholt** [comment]
Damn, awesome work!
Considered doing this myself a couple of days ago even when I tried to start migrating aw-watcher-web to aw-client-js but didn't have time to start working on it.

**Comment by Otto-AA** [comment]
I guess that's all from my side. If you know of any way to test if it fixes [ActivityWatch/aw-watcher-web#16](https://github.com/ActivityWatch/aw-watcher-web/issues/16), help on that would be appreciated. Else I think it is ready to merge...?

**Comment by johan-bjareholt** [comment]
I don't know any way to make a unittest to verify that it works, but it's clear from your implementation that this should fix it (waits for first heartbeat in queue to be received by aw-server before sending the next one). Will review properly soon

File: src/aw-client.ts [hunk 1]
```diff
@@ -90,10 +104,54 @@ class AWClient {
         return this.req.post('/0/buckets/' + bucket_id + "/events", events);
     }
 
-    heartbeat(bucket_id: string, pulsetime: number, data: Heartbeat) {
+    private send_heartbeat(bucket_id: string, pulsetime: number, data: Heartbeat) {
         return this.req.post('/0/buckets/' + bucket_id + "/heartbeat?pulsetime=" + pulsetime, data);
     }
 
+    heartbeat(bucket_id: string, pulsetime: number, heartbeat: Heartbeat): AxiosPromise {
+        if (!this.heartbeatQueues.hasOwnProperty(bucket_id)) {
```
Comment (johan-bjareholt) [inline src/aw-client.ts]: Add comment such as "Create queue for bucket if not already exists"

File: src/aw-client.ts [hunk 2]
```diff
@@ -90,10 +104,54 @@ class AWClient {
...
+            this.heartbeatQueues[bucket_id] = {
+                isProcessing: false,
+                data: []
+            };
+        }
+
```
Comment (johan-bjareholt) [inline src/aw-client.ts]: nitpicking, but preferably only one empty line here

**Comment by Otto-AA** [comment]
Do we need to do something similar for the `insertEvent` function or is this not affected by this problem?

**Comment by johan-bjareholt** [comment]
@Otto-AA Well, the recommended way is for watchers to use heartbeats so unless someone actually has some usecase for it we can skip this. I don't think we have that in the python aw-client either actually.

**Comment by johan-bjareholt** [comment]
Awesome work! :)

**Pull request merged**